frontend/src/metabase/querying/notebook/components/NotebookStepList/NotebookStepList.unit.spec.tsx

```diff
@@ -0,0 +1,166 @@
+/* eslint jest/expect-expect: ["error", { "assertFunctionNames": ["assertActionButtonsOrder"] }] */
+import type { ComponentProps } from "react";
+
+import { createMockEntitiesState } from "__support__/store";
+import { renderWithProviders, screen } from "__support__/ui";
+import { getMetadata } from "metabase/selectors/metadata";
+import Question from "metabase-lib/v1/Question";
+import type {
+  Card,
+  Filter,
+  JoinCondition,
+  StructuredQuery,
+} from "metabase-types/api";
+import {
+  createMockCard,
+  createMockStructuredDatasetQuery,
+} from "metabase-types/api/mocks";
+import {
+  ORDERS,
+  ORDERS_ID,
+  PRODUCTS,
+  PRODUCTS_ID,
+  createSampleDatabase,
+} from "metabase-types/api/mocks/presets";
+import {
+  createMockQueryBuilderState,
+  createMockState,
+} from "metabase-types/store/mocks";
+
+import { NotebookStepList } from "./NotebookStepList";
+
+const ORDERS_PRODUCT_JOIN_CONDITION: JoinCondition = [
+  "=",
+  ["field", ORDERS.PRODUCT_ID, null],
+  ["field", PRODUCTS.ID, { "join-alias": "Products" }],
+];
+
+const PRODUCTS_JOIN = {
```

Comment:
We don't use MBQL when writing unit tests now. Please use MBQL lib for that https://github.com/metabase/metabase/blob/8486a9f9c85b4820c99249468286ead200a79c9c/frontend/src/metabase/querying/filters/hooks/use-filter-modal/utils/filters.unit.spec.ts#L9

Reply:
but I use MBQL to create cards, not queries. do we have a way to create a card using metabase-lib?

Reply:
`new Question(createMockCard(), metadata).setQuery(...)`

Reply:
```
const question = Question.create({ databaseId, metadata });
question = question.setQuery(query);
```

just found it, thanks